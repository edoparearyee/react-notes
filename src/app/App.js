import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';

import './App.scss';
import TodosView from './components/TodosView/TodosView';
import AboutView from './components/AboutView/AboutView';

class App extends Component {
  render() {
    return (
      <div className="App">
        <nav>
          <ul className="App__Nav">
            <li className="App__NavItem">
              <Link to="/">Home</Link>
            </li>
            <li className="App__NavItem">
              <Link to="/about">About</Link>
            </li>
          </ul>
        </nav>
        <Route exact path="/" component={TodosView} />
        <Route path="/about" component={AboutView} />
      </div>
    );
  }
}

export default App;
