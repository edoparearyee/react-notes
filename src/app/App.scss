@import "../scss/variables";

.App {
  padding-top: 64px;
}
