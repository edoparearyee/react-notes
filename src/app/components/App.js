import React, { Component } from 'react';

import './App.scss';
import TodoListContainer from '../containers/TodoList/TodoList';
import TodoAddFormContainer from '../containers/TodoAddForm/TodoAddForm';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Todos</h1>
        <TodoListContainer />
        <TodoAddFormContainer />
      </div>
    );
  }
}

export default App;
