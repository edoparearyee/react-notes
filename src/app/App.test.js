import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

it('renders component', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
