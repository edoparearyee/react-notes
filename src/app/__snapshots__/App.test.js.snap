// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div
  className="App"
>
  <nav>
    <ul
      className="App__Nav"
    >
      <li
        className="App__NavItem"
      >
        <NavLink
          activeClassName="App__NavItem--Active"
          aria-current="page"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li
        className="App__NavItem"
      >
        <NavLink
          activeClassName="App__NavItem--Active"
          aria-current="page"
          to="/about"
        >
          About
        </NavLink>
      </li>
    </ul>
  </nav>
  <h1>
    React Notes
  </h1>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/about"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
